similarity-search/src/main/java/org/phenotips/similarity/internal/SolrSimilarPatientsFinder.java

```diff
@@ -149,36 +149,7 @@ public long countSimilarPatients(Patient referencePatient)
             }
         }
 
-        return getTopResults(results);
-    }
-
-    private List<PatientSimilarityView> getTopResults(List<PatientSimilarityView> allResults)
-    {
-        if (allResults.isEmpty()) {
-            return Collections.emptyList();
-        }
-
-        // Select top k via PriorityQueue
-        PriorityQueue<PatientSimilarityView> pq =
-            new PriorityQueue<>(allResults.size(), new Comparator<PatientSimilarityView>()
-            {
-                @Override
-                public int compare(PatientSimilarityView o1, PatientSimilarityView o2)
-                {
-                    return (int) Math.signum(o2.getScore() - o1.getScore());
-                }
-            });
-        pq.addAll(allResults);
-
-        List<PatientSimilarityView> topResults = new ArrayList<>();
-        int docSize = pq.size();
-        for (int i = 0; i < docSize; i++) {
-            PatientSimilarityView item = pq.poll();
-            if (item != null) {
-                topResults.add(item);
-            }
-        }
-        return topResults;
+        return results;
```

Comment:
There is a change in functionality now, `find()` used to return sorted results, now it does not. I know this is taken care of by the new sorting functionality in RM-pagination, but conceptually it seems better to sort here (so that all users of `find()` get sorted results) instead of higher in the code stack?

Reply:
Do all users of `find()` require sorted results? And will be sorting order  preserved down stack? Will be any matches filtered/excluded from list later? Seems better to do sorting for final results at the place of clusterview initialised.